eventsourcing/src/test/java/org/axonframework/eventsourcing/EventCountSnapshotTriggerDefinitionTest.java

```diff
@@ -70,24 +70,45 @@ void tearDown() {
     }
 
     @Test
-    void testSnapshotterTriggeredOnUnitOfWorkCommit() {
+    void testSnapshotterTriggeredOnUnitOfWorkCleanup() {
         SnapshotTrigger trigger = testSubject.prepareTrigger(aggregate.rootType());
-        GenericDomainEventMessage<String> msg = new GenericDomainEventMessage<>("type", aggregateIdentifier, (long) 0,
+        GenericDomainEventMessage<String> msg = new GenericDomainEventMessage<>("type", aggregateIdentifier, 0,
                                                                                 "Mock contents", MetaData.emptyInstance());
         trigger.eventHandled(msg);
         trigger.eventHandled(msg);
         trigger.eventHandled(msg);
         trigger.eventHandled(msg);
 
         verify(mockSnapshotter, never()).scheduleSnapshot(aggregate.rootType(), aggregateIdentifier);
+        CurrentUnitOfWork.get()
+                         .onCommit(uow -> verify(mockSnapshotter, never())
+                                 .scheduleSnapshot(aggregate.rootType(), aggregateIdentifier));
         CurrentUnitOfWork.commit();
         verify(mockSnapshotter).scheduleSnapshot(aggregate.rootType(), aggregateIdentifier);
     }
 
+    @Test
+    void testSnapshotterTriggeredOnUnitOfWorkCommit() {
+        SnapshotTrigger trigger = testSubject.prepareTrigger(aggregate.rootType());
+        GenericDomainEventMessage<String> msg = new GenericDomainEventMessage<>("type", aggregateIdentifier, 0,
+                                                                                "Mock contents", MetaData.emptyInstance());
+        trigger.initializationFinished();
+        trigger.eventHandled(msg);
+        trigger.eventHandled(msg);
+        trigger.eventHandled(msg);
+        trigger.eventHandled(msg);
+
+        verify(mockSnapshotter, never()).scheduleSnapshot(aggregate.rootType(), aggregateIdentifier);
+        CurrentUnitOfWork.get()
+                         .onCommit(uow -> verify(mockSnapshotter)
+                                 .scheduleSnapshot(aggregate.rootType(), aggregateIdentifier));
+        CurrentUnitOfWork.commit();
```

Comment:
I would change this testcase and include 2 variants.
Move the "verify" code outside of the onCommit. It's unreliable to put assertions there. If the UoW is rolled back, the test passes.
One variant should do a UOW Commit, and the other a rollback. In case of the commit there is an invocation, in case of rollback, there isn't